.github/workflows/repository_dispatch.yml

```diff
@@ -1,7 +1,31 @@
-name: Bahmni e2e tests
-
-on: repository_dispatch
+name: Bahmni E2E Tests & Repository Dispatch
 
+on: 
+  push:
+    branches:
+      - main
+  repository_dispatch:
+    types: [Smoke-Test-Dev-Lite]
+  workflow_dispatch:
+    inputs:
+      TEST_CONTEXT:
+        description: 'Test tags to be executed. eg. lite, payments, lab, hospital. Logical operators can be used.'
+        required: false
+        default: lite
+      TEST_ENV:
+        description: 'Environment to test'
+        required: true
+        type: choice
+        default: dev.lite
+        options:
+          - dev.lite
+          - qa.lite
+          - lite
+          - demo
+          - demo0.9.4
+env:
+  TEST_CONTEXT: ${{ github.event.client_payload.TEST_CONTEXT || github.event.inputs.TEST_CONTEXT || 'lite' }}
+  TEST_ENV: ${{ github.event.client_payload.TEST_ENV || github.event.inputs.TEST_ENV || 'dev.lite' }}
```

Comment:
if CI triggered then it would anyways be a dedicated env

Reply:
Yes it is if triggered by CI or on Push to main, if the workflow is triggered manually we are giving options to filter any test and to execute on any environment.

Reply:
Removed the condition.
For Push & repository Dispatch - always "dev.lite"
Manual Trigger - Any Environment